lib/iprime.man
.\" RCSID @(#)$Id: iprime.man,v 1.1 2002/07/18 22:07:50 rkowen Exp $
.\" LIBDIR
.TH "IPRIME" "3rko" "09 Sep 1999"
.SH NAME
iprime \- finds the lowest prime factor
.SH SYNOPSIS

.nf
#include "iprime.h"

         char  chprime(char n);
  signed char  scprime(signed char n);
unsigned char  ucprime(unsigned char n);
         short  hprime(short n);
unsigned short uhprime(unsigned short n);
         int    iprime(int n);
unsigned int   uiprime(unsigned int n);
         long   lprime(long n);
unsigned long  ulprime(unsigned long n);
.fi

.SH DESCRIPTION
The
.I iprime
collection returns the lowest prime factor for the given
integer argument.  It straight forwardly searches from the
first few primes until it exhausts the look-up table.
After that it then tries the odd numbers
upto the sqrt of the argument.

.SH SEE ALSO
isqrt(3rko),gcd(3rko)

.SH DIAGNOSTICS
Returns 0 for invalid input values.

.SH AUTHOR
R.K.Owen,Ph.D.

.KEY WORDS


NEWS
* Note: Implied Consent -
  If you send in a code patch or fix of one type or another you consent
  to have your name and email address listed here unless you explicitly
  note otherwise.  Whether your name & email is actually listed depends
  on how extensive the change.  (If it's a minor code, document change,
  or spelling correction your name will probably not be listed.)

****************************   Release 3.2 ******************************
Modules 3.2.9
	* Fix the coexistence of load/remove flags (Martin Siegert)
	* Fixed an overlapping string copy (Wilson Snyder & Orion Poplawski)
	* Isolated calls to the new interpreter and get & set env.vars.
	* Optionally use Tcl memory check routines for malloc/realloc.
	* Fixed the "module purge" memory corruption (Poor Yorick)
	* Added support for Ruby (Tammo Tjarks)

Modules 3.2.8
Oct  1, 2010				R.K. Owen (rk@owen.sj.ca.us)
	* Handle the .modulerc/.version in deep modulefile dirs,
	  Patches thanks to Lindsay Todd, and others.
	* Note that .version takes precedence over .modulerc files.
	* Renamed init/python to init/python.py and updated with
	  anonymous patch to eliminate 'sed' references
	* Renamed init/perl to init/perl.pm
	* Added brief section to module.1 on invoking modules for perl & python
	* Added Nathaniel Waisbrot's patch for CMake support with tests.
	* Fixed the MANPATH in modulefiles/modules to use configure generated
	  values.
	* Added the configure --with-man-path so a system specific MANPATH
	  could be used.  Some "man"s radically change their search path
	  when a MANPATH exists and when not.
	  The default is "/usr/man:/usr/share/man" (when both exists).
	* Purging a purged module list does not result in an error.
	* Fixed the configure script for searching for tclConfig.sh
	  and handle --without-x in the tests.

Modules 3.2.7
Jul 30, 2009				R.K. Owen (rk@owen.sj.ca.us)
	* Have init/bash export the module function (thanks to Roy Dragseth)
	* Fixed swap bug when prepending or appending multiple times
	  (fix thanks to Gavin Walker), added tests for same.
	* Have purge remove loaded modules in reverse order
	  (patch contributed by carriees)
	* Added a batch of patches from Gerrit Renker
	  - Goes through each of the two manpages, bringing them up to date.
	  - Fix an error in configure.ac caused by misspelling
	  - Fix typos in help messages and source-code comments
	  - Fix the setup of $MODULESPATH to not have a trailing ':'
	  - Fix segmentation violation occurring when the shelltype is absent
	  - Fix a condition where 'make uninstall' does not cleanse a directory
	  - Fix a build problem which set the executable bit on 'data' files
	  - Cleans up noise in '.modulespath' file
	  - Replaces external sed filter script with internal Makefile.am rule
	  - Adds missing 'zsh' case in etc/global
	  - simplify (t)csh initialisation
	  - Add bash auto-completion for the 'module' command
	  - Adds new 'chdir' modulefile command (sets cwd on module load)
	* versions a & b fixed some bugs

Modules 3.2.6
Oct 29, 2007				R.K. Owen (rk@owen.sj.ca.us)
	* Fixed the locate_module.c:realloc() calls which didn't properly
	  take the element size into account (Fix thanks to Ron Isaacson).
Modules 3.2.5
Feb 14, 2007				R.K. Owen (rk@owen.sj.ca.us)
	* Wrapped all malloc() calls to make sure the size is never 0.
	  This fixes a problem spotted by Francesco Nigro under AIX,
	  and diagnosed by G Lehnert.
	* Updated tests when domainname is unknown.
Modules 3.2.4
Feb  2, 2007				R.K. Owen (rk@owen.sj.ca.us)
	* Fixed bug where "empty" directory paths in the MODULEPATH env.var.
	  needed to be handled gracefully.
Modules 3.2.3
Jun  1, 2006				R.K. Owen (rk@owen.sj.ca.us)
	* Allow version names to be substrings of other version names
	  thanks to dgoldendias at users.sourceforge.net
	* Fixed compiling bug when --enable-log was specified.
	* Made some minor changes to the syslogging and tests so that
	  they all pass.
	* the set-alias command will for Bourne shell derivatives
	  substitute the safe ${1+"$@"} for $*.
	* Added the --delim option to prepend-path and append-path to
	  allow arbitrary delimiters instead of the colon ':'
	* Fixed build problem with regards to the init/* files
	  (solution due to Thomas Zeiser)
Modules 3.2.2
May  4, 2006				R.K. Owen (rk@owen.sj.ca.us)
	* Replaced all strtok() calls with an internal version.
	* Now can append/prepend and empty string '' or colons ':'
	* Added more --with-tclx-* options in case tclxConfig.sh
	  can not be found, but the TclX lib does exist.
	* Expanded the search path for tclConfig.sh.
	* Updated the TclX init call for 8.4
Modules 3.2.1
Feb 13, 2006				R.K. Owen (rk@owen.sj.ca.us)
	* Catch 'exit N' within modulefiles, use this as an exit return code.
	  Has the behavior of loading the modulefile up to the point of exit.
	* For bash, a non-zero exit will append a test that returns false.
	* Added tests for "break" and "exit N"
	* various other bug fixes
Modules 3.2.0
Jan 19, 2006				R.K. Owen (rk@owen.sj.ca.us)
	* default for cache is now "no"
	* Added version_test() to testsuite to make some of the
	  tests conditional on the values generated by --version.
	* Reworked every configure-based file
	  - the configure options help is formatted with AC_HELP_STRING
	  - uses automake to generate most of the Makefile.in
	  - renamed configure.in to configure.ac per autoconf
	  - Versioning is optional now with a configure option
	  - reworked the init scripts for versioning or not
	  - eliminated acconfig.h and put templates in configure.ac
	  - make install now honors the DESTDIR variable for package builds
	  - honors bindir for modulecmd location
	   (but the init scripts are only correct if no versioning)
	* Collected the usage info and now it can be displayed with
	  the --help or -H option (same as help)
	* --version now gives most of the optional configure parameters
	  for easy debugging of problems
	* Made the output deterministic (sorted) and reworked tests also

lib/iprime/iprime.c
static const char RCSID[]="@(#)$Id: iprime.c,v 1.1 2002/07/18 22:07:50 rkowen Exp $";
static const char AUTHOR[]="@(#)iprime 1.0 1999/09/09 R.K.Owen,Ph.D.";
/* iprime.c -
 * a simple minded routine to find out whether a positive number is prime.
 * returns 1 if prime, else it returns the smallest factor > 1 if not.
 * An invalid value causes the function to return 0;
 */
/** ** Copyright *********************************************************** **
 ** 									     **
 ** Copyright 1999 by R.K.Owen,Ph.D.		                      	     **
 ** last known email: librko@kooz.sj.ca.us				     **
 **                   rk@owen.sj.ca.us					     **
 ** see LICENSE.LGPL, which must be provided, for details		     **
 ** 									     **
 ** ************************************************************************ **/

#include "iprime/iprime_.h"

/* just like templates - almost */
_IPRIME(iprime, int, isqrt)

